# Project Setup & Data Loading

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv(r"netflix_content_intelligence_combined.csv")

# Data Cleaning & Feature Selection

In [3]:
df = df.drop_duplicates()

df = df.dropna(subset = 'date_added')

df['date_added'] = pd.to_datetime(df['date_added'], format='%d-%b-%y', errors='coerce')

fill_values = {
    'rating': 'Unrated',
    'countries': 'Unknown',
    'primary_country': 'Unknown',
    'directors': 'Unknown',
    'cast': 'Unknown'
}
df = df.fillna(value=fill_values)

In [4]:
missing = df.isnull().sum()
missing[missing > 0]

date_added                 88
runtime_minutes          2662
languages                2422
original_language        2422
imdb_id                  2181
tmdb_id                  2422
imdb_rating              2317
imdb_votes               2317
tmdb_rating              2422
tmdb_popularity          2422
imdb_rating_category     2317
tmdb_rating_category     2422
popularity_category      2422
popularity_percentile    2422
dtype: int64

# Exploratory Data Analysis (EDA)

In [5]:
# How is the distribution of movies versus TV shows,and which ones make up a larger share of Netflix?

type_count = df['type'].value_counts()
type_percent = df['type'].value_counts(normalize=True) * 100

pd.DataFrame({'Count': type_count, 'Percentage %': type_percent})

,Count,Percentage %
type,,
Movie,5376,69.135802
TV Show,2400,30.864198


Netflix dedicates approximately 70% of its archive to feature films and about 30% to series.

In [6]:
# Which countries have the most content production on Netflix?

top_countries = df[df['primary_country'] != 'Unknown']['primary_country'].value_counts().head(5)
top_countries_percent = (top_countries / len(df)) * 100

pd.DataFrame({'Count': top_countries , 'Percentage': top_countries_percent})

,Count,Percentage
primary_country,,
United States,2877,36.998457
India,956,12.294239
United Kingdom,576,7.407407
Canada,259,3.330761
Japan,236,3.034979


The United States is the most powerful content provider on the platform, accounting for more than 37 percent of the total Netflix archive. India is in second place with 12.28 percent, reflecting Netflix's special focus on the Bollywood film market. The United Kingdom, Canada, and Japan are next in line.

In [7]:
# Users give better ratings to movies or TV shows?

df.groupby('type')['imdb_rating'].agg(['mean' , 'median' , 'count']).round(2)

,mean,median,count
type,,,
Movie,6.21,6.3,3965
TV Show,6.94,7.1,1494


Although movies make up about 70 percent of Netflix's total library, series have been more popular with viewers. The average rating for series (6.94) is significantly higher than for movies (6.21).

In [8]:
# Top 10 Genres Based on Content Quantity and Quality

df[df['primary_genre'] != 'Unknown'].groupby('primary_genre').agg(count = ('type', 'count'), mean_rating = ('imdb_rating' , 'mean')).sort_values(['count' , 'mean_rating'] , ascending = False).head(10).round(2)

,count,mean_rating
primary_genre,,
Dramas,1383,6.39
Comedies,1074,5.93
Documentary,944,7.03
Action & Adventure,721,6.03
International TV Shows,689,6.95
Children & Family Movies,502,5.95
Crime TV Shows,369,7.08
Kids,357,6.55
Stand-Up Comedy,321,6.46


Although drama and comedy genres make up the majority of Netflix's library, the highest levels of user satisfaction are for documentaries (with an average of 7.03) and crime series (with an average of 7.08). In contrast, horror films, despite their significant production volume, have the lowest average score (5.20).

# Key insights

Content distribution: 69% movies and 31% series.

Quality: Series (average 6.94) have a higher score than movies (6.21).

Countries: America and India have the largest share of production.

Genres: The highest volume is related to Drama and Comedy, but the highest satisfaction is related to Documentary and Crime.

# Export Cleaned Dataset

In [10]:
df.to_csv('netflix_cleaned.csv', index=False)